core/src/main/java/org/apache/iceberg/view/BaseMetastoreViewCatalog.java

```diff
@@ -0,0 +1,217 @@
+    private final TableIdentifier identifier;
+    private final ImmutableViewVersion.Builder viewVersionBuilder = ImmutableViewVersion.builder();
+    private final Map<String, String> properties = Maps.newHashMap();
+    private final List<ViewRepresentation> representations = Lists.newArrayList();
+    private Namespace defaultNamespace = null;
+    private Schema schema = null;
+
+    protected BaseViewBuilder(TableIdentifier identifier) {
+      Preconditions.checkArgument(
+          isValidIdentifier(identifier), "Invalid view identifier: %s", identifier);
+      this.identifier = identifier;
+    }
+
+    @Override
+    public ViewBuilder withSchema(Schema newSchema) {
+      this.schema = newSchema;
+      viewVersionBuilder.schemaId(newSchema.schemaId());
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withQuery(String dialect, String sql) {
+      representations.add(
+          ImmutableSQLViewRepresentation.builder().dialect(dialect).sql(sql).build());
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withDefaultCatalog(String defaultCatalog) {
+      viewVersionBuilder.defaultCatalog(defaultCatalog);
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withDefaultNamespace(Namespace namespace) {
+      this.defaultNamespace = namespace;
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withProperties(Map<String, String> newProperties) {
+      this.properties.putAll(newProperties);
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withProperty(String key, String value) {
+      this.properties.put(key, value);
+      return this;
+    }
+
+    @Override
+    public View create() {
+      return create(newViewOps(identifier));
+    }
+
+    @Override
+    public View replace() {
+      return replace(newViewOps(identifier));
+    }
+
+    @Override
+    public View createOrReplace() {
+      ViewOperations ops = newViewOps(identifier);
+      if (null == ops.current()) {
+        return create(ops);
+      } else {
+        return replace(ops);
+      }
+    }
+
+    private View create(ViewOperations ops) {
+      if (null != ops.current()) {
+        throw new AlreadyExistsException("View already exists: %s", identifier);
+      }
+
+      Preconditions.checkState(
+          !representations.isEmpty(), "Cannot create view without specifying a query");
+      Preconditions.checkState(null != schema, "Cannot create view without specifying schema");
+      Preconditions.checkState(
+          null != defaultNamespace, "Cannot create view without specifying a default namespace");
+
+      ViewVersion viewVersion =
+          viewVersionBuilder
+              .versionId(1)
+              .addAllRepresentations(representations)
+              .defaultNamespace(defaultNamespace)
+              .timestampMillis(System.currentTimeMillis())
+              .putSummary("operation", "create")
+              .build();
```

Comment:
This doesn't set `defaultCatalog`, which looks like a bug to me. In addition, the `withDefaultCatalog` method above sets the default catalog on a `viewVersionBuilder` that is unused. It looks like this wasn't updated after originally using a builder?

Reply:
`withDefaultCatalog()` sets it on the `viewVersionBuilder` that is being used in L151 for creating a view version and then also in L194. 
We do have tests that check for the default catalog when parsing view metadata. Additionally, `ViewCatalogTests#replaceViewVersion()` was setting + testing the default catalog (and now also `ViewCatalogTests#completeCreateView()`).